zrml/court/src/lib.rs

```diff
@@ -16,611 +16,2306 @@
+        /// When this is called the court participant is not anymore able to get drawn for new cases.
+        /// The court participant gets removed from the stake-weighted pool.
+        /// After that the court participant can exit the court.
+        ///
+        /// # Weight
+        ///
+        /// Complexity: `O(log(n))`, where `n` is the number of jurors in the stake-weighted pool.
+        #[pallet::weight(T::WeightInfo::prepare_exit_court(T::MaxCourtParticipants::get()))]
+        #[transactional]
+        pub fn prepare_exit_court(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+
+            let mut prev_p_info =
+                <Participants<T>>::get(&who).ok_or(Error::<T>::JurorDoesNotExist)?;
+            ensure!(prev_p_info.prepare_exit_at.is_none(), Error::<T>::AlreadyPreparedExit);
+
+            let mut pool = CourtPool::<T>::get();
+            let pool_len = pool.len() as u32;
+
+            // do not error in the else case
+            // because the juror might have been already removed from the pool
+            if let Some((index, _)) = Self::get_pool_item(&pool, prev_p_info.stake, &who) {
+                pool.remove(index);
+                <CourtPool<T>>::put(pool);
+            }
+
+            let now = <frame_system::Pallet<T>>::block_number();
+            prev_p_info.prepare_exit_at = Some(now);
+            <Participants<T>>::insert(&who, prev_p_info);
+
+            Self::deposit_event(Event::ExitPrepared { court_participant: who });
+
+            Ok(Some(T::WeightInfo::prepare_exit_court(pool_len)).into())
+        }
+
+        /// Exit the court.
+        /// The stake which is not locked by any court case is unlocked.
+        /// `prepare_exit_court` must be called before
+        /// to remove the court participant (juror or delegator) from the stake-weighted pool.
+        ///
+        /// # Arguments
+        ///
+        /// - `court_participant`: The court participant,
+        /// who is assumed to be not be part of the pool anymore.
+        ///
+        /// # Weight
+        ///
+        /// Complexity: `O(log(n))`, where `n` is the number of jurors in the stake-weighted pool.
+        #[pallet::weight(T::WeightInfo::exit_court_set().max(T::WeightInfo::exit_court_remove()))]
+        #[transactional]
+        pub fn exit_court(
+            origin: OriginFor<T>,
+            court_participant: AccountIdLookupOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?;
+
+            let who = T::Lookup::lookup(court_participant)?;
+
+            let mut prev_p_info =
+                <Participants<T>>::get(&who).ok_or(Error::<T>::JurorDoesNotExist)?;
+
+            let now = <frame_system::Pallet<T>>::block_number();
+            let prepare_exit_at =
+                prev_p_info.prepare_exit_at.ok_or(Error::<T>::PrepareExitAtNotPresent)?;
+            ensure!(
```

Comment:
I agree, no good solution in mind at the moment. But just wanted to mention it is far better than not having a lock period in place like before.

Reply:
By the way, I just realised that the current lock period makes no sense at all, because whenever the juror leaves the court pool, he doesn't take the risk of being drawn anymore. I strongly need to reconsider the lock period.

Reply:
Although... It does make sense so that hopping is disincentivised. 

The only part which is missing, is distributing inflation only after you were already a participant of the court system for at least one inflation period. For this I need to store the block number when the participant joined the court.